### PR TITLE
:running: Add test for typed Patch after Status.Patch

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

This PR adds a simple test to make sure that running Patch after Status.Patch.

### CODE REVIEW

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -1178,6 +1178,34 @@ var _ = Describe("Client", func() {
 				close(done)
 			})
 
+			It("should patch correctly after a status patch", func(done Done) {
```
Comment (DirectXMan12) [inline pkg/client/client_test.go]: err... what does "correctly" mean here?  I feel like I'm missing some context for this.

Comment (vincepri) [reply pkg/client/client_test.go]: Apologies, we've noticed that the other way around, doing Status.Patch after Patch makes the object lose status information. I added this test to ensure that, going forward, this behavior is correct.

I can also add the opposite (failing) test and mark it to be Skipped if you prefer

Comment (DirectXMan12) [reply pkg/client/client_test.go]: ah, I think you should just elaborate a bit in the test description

Comment (vincepri) [reply pkg/client/client_test.go]: Sounds good, I reworded a little yesterday but can add more comments.

FYI, we ended up adding a package to wrap the Patch logic for us https://github.com/kubernetes-sigs/cluster-api/pull/1275

**Comment by DirectXMan12** [comment]
hmm... the problem is that what you're describing is kinda intended behavior, but is a bit of a footgun :-/.  I think it kinda makes sense that if you update status, you only expect the status bits to be filled in from the server.

**Comment by vincepri** [comment]
Definitely agree, do you think it might be worth discussing in a separate issue or should we leave as it is and maybe add only some docs?

**Comment by DirectXMan12** [comment]
let's discuss in a separate issue

**Comment by DirectXMan12** [comment]
I think the conclusion was that we could close this

**Pull request closed without merging**